crates/sui-surfer/src/surf_strategy.rs

```diff
@@ -7,7 +7,7 @@ use tokio::sync::watch;
 use crate::surfer_state::{EntryFunction, SurferState};
 
 #[async_trait]
-pub trait SurfStrategy: Send + Sync + 'static {
+pub trait SurfStrategy: Send + Sync + Clone + 'static {
```

Comment:
So Box<dyn> and Clone don't really go together very well. I bet thats whats causing things to choke. use an Arc instead or get rid of the dyn.